### PR TITLE
feat: add Report

### PR DESCRIPTION
**Pull request by MahavishnuT**
Hi 👋 

# OVERVIEW

I added the report page for your website. This permits a user to report a sticker pack if found offensive or contrary to the rules.

+ Uses the API route `/report/`
+ Opens a new page, not a pop-up for the report itself

### CODE REVIEW

File: src/components/report/Report.tsx [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+                      </small>
+                      <div className="invalid-feedback">
+                        <ErrorMessage name="secAnswer" />
+                        &nbsp;
+                      </div>
+                    </label>
+                  </div>
+                </div>
+
+                {/* [Control] Submit and Reset */}
```
Comment (romainricard) [inline src/components/report/Report.tsx:301]: Remove `and Reset`

File: src/components/report/Report.tsx [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Beatae
+            pariatur sequi, sint consectetur fugiat neque blanditiis quas id
+            iusto mollitia at libero quis eos velit soluta. Nobis amet, corrupti
+            impedit eveniet eligendi nostrum ipsum ex doloribus omnis est
+            magnam, beatae sed voluptates sapiente consectetur aliquid
+            voluptatum iure eos inventore. Pariatur aspernatur dolores optio
+            consequatur odio! Iure necessitatibus minus sapiente! Aliquam
+            quibusdam recusandae deleniti rerum quas, et reprehenderit odit
+            minima, corporis, doloribus ratione ipsam qui. Numquam inventore
+            quibusdam veniam voluptatem dolor?
```
Comment (romainricard) [inline src/components/report/Report.tsx:207]: Use the following text:

```
<p>Use this form to report a pack to Signalstickers' admins.</p>
<p>Please add context and explain why you are reporting this pack.</p>
```

**Review by romainricard — CHANGES_REQUESTED** [review]
Looks good, thanks for the feature!
Two points are missing:
- Install eslint in your IDE and fix the formating (including whitespaces added by your linter)
- Fix the review comments :pray:

File: src/components/report/Report.tsx (hunk 2 above)
Comment (MahavishnuT) [reply src/components/report/Report.tsx:207]: 👍

File: src/components/report/Report.tsx [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+    React.useState('');
+  const [reportRequestQuestion, setReportRequestQuestion] =
+    React.useState('');
+  const { packId } = useParams<UrlParams>();
+  // StickerPack object for the requested pack.
+  const [stickerPack, setStickerPack] = React.useState<StickerPack>();
+  // One of many possible error codes we may catch when trying to load or
+  // decrypt a sticker pack. This will be used to determine what error message
+  // to show the user.
+  const [stickerPackError, setStickerPackError] = React.useState('');
```
Comment (romainricard) [inline src/components/report/Report.tsx:79]: This has become useless, please remove

File: src/components/layout/AppLayout.tsx
Comment (romainricard) [inline src/components/layout/AppLayout.tsx:1]: Still some formating problems (spaces at the beginning of lines)

File: src/components/pack/StickerPackDetail.tsx
Comment (romainricard) [inline src/components/pack/StickerPackDetail.tsx:1]: Still some formating problems (spaces at the beginning of lines)

**Review by romainricard — CHANGES_REQUESTED** [review]

**Review by romainricard — APPROVED** [review]

**Comment by romainricard** [comment]
Thanks a lot @MahavishnuT! :sparkles:

**Pull request merged**